### ISSUE TITLE
Wrong week number for months range

### ISSUE THREAD

**Issue by PixelVibe** [gf3/moment-range#72]
Hi

I am having a strange problem when trying to get the week numbers of a range of months.
![screen shot 2015-04-17 at 10 55 39](https://cloud.githubusercontent.com/assets/1628148/7198879/54b65a52-e4f0-11e4-8d00-56fd5cf0fb6a.png)

This is not happening always, and it depends on how many months apart is the range. For example if the past is 1, 2 or 4 months, I am having the correct isoWeek for today. If the past is 3, 5 or 6 (the max I have tested it), the range seems to return the wrong isoWeek.

Any ideas?

Thanks

**Comment by gf3** [comment]
hmm, i wonder if it's related to the fact that the start and end dates are in different time zones. i'll try and reproduce when i have some time

**Comment by gf3** [comment]
looks like it's because of the way iteration works. the final date (`today`) is _less than_ one week after the last iteration. check out the value of `n` in the following:

``` javascript
moment = require('moment'); undefined
// undefined
require('./lib/moment-range'); undefined
// undefined
a = moment('2015-01-17T10:50:04+01:00'); undefined
// undefined
b = moment('2015-04-17T10:29:55+02:00'); undefined
// undefined
r = moment.range(a, b); undefined
// undefined
r.by('week', function(i) { console.log(i.format()); }); undefined
2015-01-17T04:50:04-05:00
2015-01-24T04:50:04-05:00
2015-01-31T04:50:04-05:00
2015-02-07T04:50:04-05:00
2015-02-14T04:50:04-05:00
2015-02-21T04:50:04-05:00
2015-02-28T04:50:04-05:00
2015-03-07T04:50:04-05:00
2015-03-14T04:50:04-04:00
2015-03-21T04:50:04-04:00
2015-03-28T04:50:04-04:00
2015-04-04T04:50:04-04:00
2015-04-11T04:50:04-04:00
// undefined
n = moment('2015-04-11T04:50:04-04:00').add(1, 'week'); undefined
// undefined
n.format()
// '2015-04-18T04:50:04-04:00'
n.isoWeek()
// 16
r.contains(n)
// false
```

**Comment by davidjrice** [comment]
So I've had the exact same problem in outputting a calendar list of weeks. I tried to find bugs in your **DateRange** code (sorry) but I couldn't find any ;) or see even a potential fix.

I think what this comes down to is that iterating over calendar weeks is more of a **special case** and generally revolves around what **weekday** the start of the range falls on.

I solved the problem for my needs by adding / subtracting the weekday from the start and end of the actual range I want to work with, then create the range and iterate by week.
